src/iframe/setup.ts

```diff
@@ -64,6 +67,31 @@ export function registerIFrameListener(config: Config, store: Store, renderer: C
   return () => window.removeEventListener('message', receiveMessage);
 }
 
+/**
+ * Look for the actual output content inside the full width divs of the OutputArea,
+ * which may have a fixed width. If not then return undefined rather than full width.
+ *
+ * @param entry the ResizeObserverEntry containing the target element
+ * @returns
+ */
+function getWidthOfWidestOutput(entry: ResizeObserverEntry) {
+  const outputs = entry.target.querySelectorAll('.jp-OutputArea-output');
+  let outputWidth = 0;
+  outputs.forEach((output) => {
+    let widestChild = 0;
+    for (let i = 0; i < output.children.length; i++) {
+      const item = output.children.item(i);
+      if (item && item.clientWidth > widestChild) {
+        widestChild = item.clientWidth;
+      }
+    }
+    if (widestChild > outputWidth) {
+      outputWidth = widestChild;
+    }
+  });
+  return outputWidth > 0 ? outputWidth : null;
+}
+
```

Comment:
this is the meaningful bit, a lot of the upper divs in the OutputArea are full width. So we've got to take a look inside for fixed-width outputs. We fall back to `null` allowing the client to easily revert to `100%` when it can't be figured out, instead of giving them the container size